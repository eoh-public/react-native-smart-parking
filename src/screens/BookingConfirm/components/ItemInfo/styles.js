import { StyleSheet } from 'react-native';

import { Colors } from '../../../../configs';

export const styles = StyleSheet.create({
  container: {
    padding: 16,
    paddingBottom: 24,
  },
  textTitle: {
    marginBottom: 16,
  },
  box: {
    height: 48,
    borderRadius: 5,
    borderWidth: 1,
    borderColor: Colors.Gray4,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  textInfo: {
    paddingLeft: 18,
  },
});
