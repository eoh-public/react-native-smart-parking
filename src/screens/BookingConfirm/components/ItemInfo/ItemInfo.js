import React from 'react';
import { View } from 'react-native';

import Text from '../../../../commons/Text';
import { Colors } from '../../../../configs';
import { styles } from './styles';

const ItemInfo = ({ title, info }) => {
  return (
    <View style={styles.container}>
      <Text semibold type="H4" color={Colors.Black} style={styles.textTitle}>
        {title}
      </Text>
      <View style={styles.box}>
        <Text color={Colors.Gray9} style={styles.textInfo} type="H4">
          {info}
        </Text>
      </View>
    </View>
  );
};

export default ItemInfo;
