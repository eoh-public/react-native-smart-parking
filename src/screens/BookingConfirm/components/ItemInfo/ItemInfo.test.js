import React from 'react';
import { create, act } from 'react-test-renderer';
import ItemInfo from './ItemInfo';

describe('Test ItemInfo ', () => {
  let tree;

  test('render item info with plate number value', async () => {
    const plate_number = 'plate_number';
    act(() => {
      tree = create(<ItemInfo title={'license_plate'} info={plate_number} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
