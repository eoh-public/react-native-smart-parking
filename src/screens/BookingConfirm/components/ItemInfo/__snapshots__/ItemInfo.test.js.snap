// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ItemInfo render item info with plate number value 1`] = `
<View
  style={
    Object {
      "padding": 16,
      "paddingBottom": 24,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "SFProDisplay-Semibold",
          "fontSize": 16,
          "lineHeight": 24,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 16,
        },
      ]
    }
  >
    license_plate
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#E8E8E8",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#262626",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "paddingLeft": 18,
          },
        ]
      }
    >
      plate_number
    </Text>
  </View>
</View>
`;
